os_stub/spdm_device_secret_lib_sample/lib.c

```diff
@@ -266,21 +266,33 @@ bool libspdm_gen_csr(uint32_t base_hash_algo, uint32_t base_asym_algo, bool *nee
 
     uint8_t *cached_req_info;
     size_t cached_req_info_length;
+    uint8_t *cached_csr;
 
     /*device gen csr need reset*/
     if (*need_reset) {
         result = libspdm_read_cached_requester_info(base_asym_algo,
                                                     &cached_req_info, &cached_req_info_length);
 
-        /*get the cahed requester info and csr*/
+        /*get the cached requester info and csr*/
         if ((result) &&
             (cached_req_info_length == requester_info_length) &&
             (libspdm_const_compare_mem(cached_req_info, requester_info, requester_info_length)) &&
-            (libspdm_read_cached_csr(base_asym_algo, csr_pointer, csr_len))) {
+            (libspdm_read_cached_csr(base_asym_algo, &cached_csr, csr_len))) {
+
+            /*get and save cached csr*/
+            libspdm_copy_mem(*csr_pointer, csr_buffer_size, cached_csr, *csr_len);
```

Comment:
You cannot do the copy directly.

You can compare the size and return BUFFER_TOO_SMALL, if the destination is not big enough.

Reply:
Yes. I have added the check and deleted useless parameter :csr_buffer_size.